src/common/util/approximate-size.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Calculates the approximate size of a given object if it were to be serialized, including all nested properties.
+ * Estimates size but for the sake of speed does not stringify or otherwise serialize the object.
+ *
+ * @param {Object} object - The object for which size will be calculated.
+ * @returns {number} The approximate size of the object.
+ */
+export function approximateSize (object) {
+  const seen = new WeakSet()
+  const stack = [object]
+  let size = 0
+  while (stack.length) {
+    const value = stack.pop()
```

Comment:
You may want to wrap the innards of the while loop in a `try..catch` to prevent errors from this function possibly breaking a customer's site.

```javascript
while (stack.length) {
  try {
    ...
  } catch (_) {
    continue
  }
}
```